library/x509_crt.c

```diff
@@ -1102,35 +1105,30 @@ int mbedtls_x509_crt_parse_file( mbedtls_x509_crt *chain, const char *path )
     return( ret );
 }
 
-int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
+int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const MBEDTLS_TCHAR *path )
 {
     int ret = 0;
 #if defined(_WIN32) && !defined(EFIX64) && !defined(EFI32)
     int w_ret;
-    WCHAR szDir[MAX_PATH];
-    char filename[MAX_PATH];
-    char *p;
-    size_t len = strlen( path );
+    MBEDTLS_TCHAR szDir[MAX_PATH];
+    MBEDTLS_TCHAR filename[MAX_PATH];
+    MBEDTLS_TCHAR *p;
+    size_t len = mbedtls_strlen( path );
 
-    WIN32_FIND_DATAW file_data;
+    WIN32_FIND_DATA file_data;
     HANDLE hFind;
 
     if( len > MAX_PATH - 3 )
         return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
 
     memset( szDir, 0, sizeof(szDir) );
-    memset( filename, 0, MAX_PATH );
-    memcpy( filename, path, len );
-    filename[len++] = '\\';
+    memset( filename, 0, sizeof(filename) );
+    _tcsncpy( filename, path, len );
+    filename[len++] = MBEDTLS__T('\\');
     p = filename + len;
-    filename[len++] = '*';
+    filename[len++] = MBEDTLS__T('*');
 
-    w_ret = MultiByteToWideChar( CP_ACP, 0, filename, (int)len, szDir,
-                                 MAX_PATH - 3 );
-    if( w_ret == 0 )
-        return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
-
-    hFind = FindFirstFileW( szDir, &file_data );
+    hFind = FindFirstFile( szDir, &file_data );
```

Comment:
#1029 also adds an abstraction for that. Please investigate how to reconcile this change. It might be best to base this PR from #1029, but beware that #1029 may still change before it gets merged.

Reply:
Before trying to reconcile, it would be nice to get a compilable code in #1029.
I left a comment there, awaiting for changes.

Reply:
> #1029 also adds an abstraction for that.

The Windows-specific part of the `mbedtls_x509_crt_parse_path` function does not use any abstractions. It could be verified using cross-repository compare that this part of code has no differences with the current `development` branch.
Even more, there might be no gain in enforcing Unix-style abstraction here.